.github/workflows/CI-CD.yml

```diff
@@ -0,0 +1,46 @@
+name:  CI/CD
+
+on: [push]
```

Comment:
If the version is the same but we want to publish new release we will get error, so I suggest using something like this
https://stackoverflow.com/a/61892639/4650625
to just trigger publish if we pushed a new tag,  and if we create tag with npm ( something like `npm version major |minor | patch`  the tag pattern would be something like `v*.*.*` see https://github.com/Giveth/impact-graph/tags

PS: maybe we can have to github action file ( I haven't done that before myself), one file for running tests and we use it for every push , and the other  one for just pushing semantic version tags

Reply:
I think since the publish only works when pushing to main this should be fine. Branches should run tests without issue.

Ill add the tags part now.

Reply:
Thanks, fixed by adding two ymls